src/linker/Linker/XmlCustomAttributeSource.cs

```diff
@@ -362,66 +391,54 @@ void ProcessMethodSignature (TypeDefinition type, string signature, XPathNodeIte
-				method, iterator.Current.SelectChildren ("parameter", string.Empty));
-			ArrayBuilder<ArrayBuilder<Attribute>> returnParameterAnnotations = ProcessReturnParameters (type,
-				method, iterator.Current.SelectChildren ("return", string.Empty));
-
-			var parameterAnnotation = new ArrayBuilder<(string ParamName, DynamicallyAccessedMemberTypes Annotation)> ();
-			DynamicallyAccessedMemberTypes returnAnnotation = 0;
-
-			if (parameterAnnotations.Count > 0) {
-				foreach (var parameter in parameterAnnotations.ToArray ()) {
-					DynamicallyAccessedMemberTypes paramAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (parameter.Item2, _context, _xmlDocumentLocation);
-					if (paramAnnotation != 0)
-						parameterAnnotation.Add ((parameter.Item1, paramAnnotation));
-				}
-			}
-
-			if (returnParameterAnnotations.Count == 1) {
-				foreach (var returnparameter in returnParameterAnnotations.ToArray ()) {
-					DynamicallyAccessedMemberTypes returnparamAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (returnparameter, _context, _xmlDocumentLocation);
-					if (returnparamAnnotation != 0)
-						returnAnnotation = returnparamAnnotation;
-				}
-			} else {
-				_context.LogMessage (MessageContainer.CreateWarningMessage (_context,
-					$"There is more than one return parameter specified for '{method.Name}' in '{_xmlDocumentLocation}'", 2023, _xmlDocumentLocation));
-			}
-			if (returnAnnotation != 0 || parameterAnnotation.Count > 0)
-				_methods[method] = new AnnotatedMethod (returnAnnotation, parameterAnnotation.ToArray ());
-
+			IEnumerable<CustomAttribute> attributes = ProcessAttributes (iterator.Current);
+			if (attributes.Count () > 0)
+				_attributes[method] = attributes;
+			ProcessReturnParameters (method, iterator);
+			ProcessParameters (method, iterator);
 		}
 
-		ArrayBuilder<(string, ArrayBuilder<Attribute>)> ProcessParameters (TypeDefinition type,
-			MethodDefinition method, XPathNodeIterator iterator)
+		void ProcessParameters (MethodDefinition method, XPathNodeIterator iterator)
 		{
-			var methodParameters = new ArrayBuilder<(string, ArrayBuilder<Attribute>)> ();
+			iterator = iterator.Current.SelectChildren ("parameter", string.Empty);
 			while (iterator.MoveNext ()) {
-				methodParameters.Add ((GetAttribute (iterator.Current, "name"), ProcessAttributes (iterator)));
+				IEnumerable<CustomAttribute> attributes = ProcessAttributes (iterator.Current);
+				if (attributes.Count () > 0) {
+					string paramName = GetAttribute (iterator.Current, "name");
+					foreach (ParameterDefinition parameter in method.Parameters) {
+						if (paramName == parameter.Name) {
+							if (_attributes.ContainsKey (parameter))
+								_context.LogMessage (MessageContainer.CreateWarningMessage (_context, $"There are duplicate parameter names for '{paramName}' inside '{method.Name}' in '{_xmlDocumentLocation}'", 2024, _xmlDocumentLocation));
```

Comment:
Why is it warning?

Reply:
We found two values for the same thing - we'll pick one of them, but drop the other. I think this should be a warning (since we're warning about other issues in the XML already)

Reply:
What if we instead of warning we do something like `_attributes[parameter].Add(attributes);`